### PR TITLE
Hmi sends request with string param with invalid characters

### PR DESCRIPTION
**Pull request by ZhdanovP**
Fixes [#1862](https://github.com/smartdevicelink/sdl_core/issues/1862) 

This PR is **ready** for review.

### Risk
This PR makes **minor** API changes.

### Testing Plan
Provide ATF tests

### Summary
According to the issues when HMI sends request with String param with invalid characters we need to log corresponding error internally and respond with 'INVALID_DATA' to HMI.
The following RPCs do not have string params:

- SDL.ActivateApp
- SDL.GetListOfPermissions
- SDL.UpdateSDL
- SDL.GetStatusUpdate
- SDL.GetURLs

we do not have any changes for them

For RPC SDL.GetUserFriendlyMessage was added desired behaviour

OpenSDL do not consist SDL.GetDeviceConnectionStatus  RPC, so this pull request add this RPC and expected behaviour for it when HMI sends request with String param with invalid characters.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/include/sdl_rpc_plugin/commands/hmi/get_device_connection_status_request.h [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  bool IsSyntaxWrong(const std::string& str) const;
+};
+
+/**
+ * @brief Functor to retrieve all the data from the "device_data"
+ * section of the policy table and store <ID>, <name> of device,
+ * <usb_transport_status>, <transport_type> as array with key <device>
+ * in the message
+ */
+class DeviceDataRetriever {
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/include/sdl_rpc_plugin/commands/hmi/get_device_connection_status_request.h]: @ZhdanovP I think you could move this class under the private section of `GetDeviceConnectionStatusRequest` as this is just an implementation detail

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/get_device_connection_status_request.cc [hunk 2]
```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright (c) 2016, Ford Motor Company
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/get_device_connection_status_request.cc]: @ZhdanovP 2018

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/get_device_connection_status_request.cc [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+  bool operator()(const char c) const {
+    using namespace helpers;
+    return Compare<char, EQ, ONE>(c, '\n', '\t', ' ');
+  }
+};
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "DeviceDataRetriever")
+
+void DeviceDataRetriever::operator()(const std::string& device_id) {
+  FillDataForDeviceAtIndex(device_id, index_);
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/get_device_connection_status_request.cc]: @ZhdanovP 
```
 FillDataForDeviceAtIndex(device_id, index_++);
```

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/get_device_connection_status_request.cc [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+
+      std::string device_id;
+      if (!GetMandatoryParameters(current_device, device_id)) {
+        LOG4CXX_WARN(logger_,
+                     "Invalid data for device: " << device_id << " found");
+        SendErrorResponse(
+            correlation_id(),
+            static_cast<hmi_apis::FunctionID::eType>(function_id()),
+            hmi_apis::Common_Result::INVALID_DATA,
+            "");
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/get_device_connection_status_request.cc]: @ZhdanovP could you omit the last parameter in `SendErrorResponse`?

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/get_device_connection_status_request.cc [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+    return false;
+  }
+  out_value = device_data[key].asString();
+  // APPLINK-28279 - Invalidate HMI request with at least one empty String
+  // parameter
+  if (out_value.empty()) {
+    LOG4CXX_WARN(logger_, "Invalid data: Key " << key << " empty.");
+    return false;
+  }
+  if (IsSyntaxWrong(out_value)) {
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/get_device_connection_status_request.cc]: @ZhdanovP what about case when string contains not only invalid characters? I don't think that string like `aaa\n\taaa` is correct

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/on_device_connection_status_notification.cc [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+    std::string& out_device_id,
+    hmi_apis::Common_UserSetting::eType& out_usb_transport_status) const {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  int32_t transport_type;
+  int32_t usb_transport_status;
+
+  bool all_key_is_valid = true;
+  all_key_is_valid &=
+      GetKeyStringValue(device_data, strings::id, out_device_id);
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/on_device_connection_status_notification.cc]: @ZhdanovP suggest to use local temp variable for output param here and in the end of the function, if everything is correct, assign to output variables local ones as it's done for `out_usb_transport_status`

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/hmi_command_factory.cc [hunk 7]
```diff
@@ -547,6 +550,9 @@ CommandCreator& HMICommandFactory::get_creator_factory(
     case hmi_apis::FunctionID::BasicCommunication_OnAppActivated: {
       return factory.GetCreator<commands::OnAppActivatedNotification>();
     }
+    case hmi_apis::FunctionID::SDL_GetDeviceConnectionStatus: {
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/hmi_command_factory.cc]: @ZhdanovP I see notification, but where is `GetDeviceConnectionStatusRequest` and `GetDeviceConnectionStatusResponse`?

File: src/components/application_manager/src/commands/command_impl.cc [hunk 8]
```diff
@@ -214,5 +214,24 @@ DEPRECATED void CommandImpl::ReplaceHMIByMobileAppId(
   }
 }
 
+bool CommandImpl::CheckSyntax(const std::string& str,
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/src/commands/command_impl.cc]: @ZhdanovP could be const

File: src/components/application_manager/src/message_helper/message_helper.cc [hunk 9]
```diff
@@ -1490,6 +1490,9 @@ bool MessageHelper::CreateDeviceInfo(
   device_info_map[strings::transport_type] =
       app_mngr.GetDeviceTransportType(transport_type);
 
+  device_info_map[strings::usb_transport_status] =
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/src/message_helper/message_helper.cc]: @ZhdanovP style check?

File: src/components/policy/policy_external/src/policy_manager_impl.cc [hunk 10]
```diff
@@ -894,1138 +922,1152 @@ void PolicyManagerImpl::SetDeviceInfo(const std::string& device_id,
...
-      if (it->group_alias == it_curr->group_alias &&
-          it->group_id == it_curr->group_id) {
-        permissions_to_set.group_permissions.push_back(*it);
+                             device_info.connection_type,
+                             device_info.usb_transport_status)) {
+                               LOG4CXX_WARN(logger_, "Can't set device data.");
+                             }
+  }
+
+  PermissionConsent PolicyManagerImpl::EnsureCorrectPermissionConsent(
```
Comment (AKalinich-Luxoft) [inline src/components/policy/policy_external/src/policy_manager_impl.cc]: @ZhdanovP is it possible to move out formatting changes into the separate pull request? It's quite hard to review this file with all changes together

File: src/components/policy/policy_regular/src/policy_table/enums.cc [hunk 11]
```diff
@@ -3,6 +3,41 @@
 
 namespace rpc {
 namespace policy_table_interface_base {
+bool IsValidEnum(UserSetting val) {
```
Comment (AKalinich-Luxoft) [inline src/components/policy/policy_regular/src/policy_table/enums.cc]: @ZhdanovP please one more blank line after namespace

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/include/sdl_rpc_plugin/commands/hmi/get_device_connection_status_request.h (hunk 1 above)
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/include/sdl_rpc_plugin/commands/hmi/get_device_connection_status_request.h]: @AKalinich-Luxoft yes this class is already in private

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/get_device_connection_status_request.cc (hunk 4 above)
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/get_device_connection_status_request.cc]: @AKalinich-Luxoft it should be empty string

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/on_device_connection_status_notification.cc (hunk 6 above)
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/on_device_connection_status_notification.cc]: @AKalinich-Luxoft For define the out_device_id it's only one assign operation on line 131, I agree with your approach but think in this case it's redundant

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/get_device_connection_status_request.cc (hunk 5 above)
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/get_device_connection_status_request.cc]: @AKalinich-Luxoft yes this is incorrect one, but we are checking all characters in the srting by InvalidCharacterChecker for reject such cases

File: src/components/policy/policy_regular/src/policy_table/enums.cc (hunk 11 above)
Comment (ZhdanovP) [reply src/components/policy/policy_regular/src/policy_table/enums.cc]: @AKalinich-Luxoft  fixed, please see updated commits

File: src/components/application_manager/include/application_manager/commands/command_impl.h [hunk 12]
```diff
@@ -138,6 +138,14 @@ class CommandImpl : public Command {
   */
   void SetAllowedToTerminate(const bool allowed) OVERRIDE;
 
+  /**
+   * @brief Check syntax of string from mobile
```
Comment (AByzhynar) [inline src/components/application_manager/include/application_manager/commands/command_impl.h]: @ZhdanovP Does this method checks only mobile messages?

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/include/sdl_rpc_plugin/commands/hmi/get_device_connection_status_request.h [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef SRC_COMPONENTS_APPLICATION_MANAGER_INCLUDE_APPLICATION_MANAGER_COMMANDS_HMI_GET_DEVICE_CONNECTION_STATUS_REQUEST_H_
```
Comment (AByzhynar) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/include/sdl_rpc_plugin/commands/hmi/get_device_connection_status_request.h]: @ZhdanovP Wrong header guards

File: src/components/application_manager/include/application_manager/policies/policy_handler.h [hunk 14]
```diff
@@ -177,6 +177,37 @@ class PolicyHandler : public PolicyHandlerInterface,
...
+
+  /**
+   * @brief Gets IDs for devices in policy table
+   * @return Container with devices IDs
+   */
+  std::vector<std::string> GetDevicesIDs() const OVERRIDE;
+
+  /**
+   * @brief Gets USB transport status
+   * @return True if transport is enabled.
```
Comment (AByzhynar) [inline src/components/application_manager/include/application_manager/policies/policy_handler.h]: @ZhdanovP Return value is wrong

File: src/components/application_manager/include/application_manager/policies/policy_handler.h [hunk 15]
```diff
@@ -177,6 +177,37 @@ class PolicyHandler : public PolicyHandlerInterface,
...
+   */
+  hmi_apis::Common_UserSetting::eType GetDeviceUSBTransportStatus(
+      const std::string& device_id) const OVERRIDE;
+
+  /**
+   * @brief Updates USB status of mobile device in Policy Table.
+   * @param device_id Generated or obtained id of device
+   * @param usb_transport_status Status of the USB transport
+   * @return bool Success of operation
```
Comment (AByzhynar) [inline src/components/application_manager/include/application_manager/policies/policy_handler.h]: @ZhdanovP this method is void

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/include/sdl_rpc_plugin/commands/hmi/get_device_connection_status_request.h [hunk 16]
```diff
@@ -0,0 +1,160 @@
...
+   **/
+  virtual ~GetDeviceConnectionStatusRequest();
+
+  /**
+   * @brief Execute command
+   **/
+  void Run() OVERRIDE;
+
+ private:
+  DISALLOW_COPY_AND_ASSIGN(GetDeviceConnectionStatusRequest);
```
Comment (AByzhynar) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/include/sdl_rpc_plugin/commands/hmi/get_device_connection_status_request.h]: @ZhdanovP Should be the last in private section

File: src/components/application_manager/include/application_manager/commands/command_impl.h (hunk 12 above)
Comment (ZhdanovP) [reply src/components/application_manager/include/application_manager/commands/command_impl.h]: @AByzhynar yes, now only from mobile

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/hmi/get_device_connection_status_request_test.cc [hunk 17]
```diff
@@ -0,0 +1,443 @@
...
+      .WillOnce(DoAll(SetArgPointee<1u>(kDeviceName2), Return(0)));
+
+  EXPECT_CALL(mock_connection_handler_, GetDeviceID(kDeviceId1, _))
+      .WillOnce(DoAll(SetArgPointee<1>(kDeviceHandle1), Return(true)));
+  EXPECT_CALL(mock_connection_handler_, GetDeviceID(kDeviceId2, _))
+      .WillOnce(DoAll(SetArgPointee<1>(kDeviceHandle2), Return(true)));
+
+  SetSuccessfulDataRetrievalExpectations();
+
+  command->Run();
```
[the remark this answers is not in the document]
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/hmi/get_device_connection_status_request_test.cc]: @AByzhynar we didn't overload the RequestFromHMI::Init for GetDeviceConnectionStatusRequest

[the remark this answers is not in the document]
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/hmi/get_device_connection_status_request_test.cc]: @AByzhynar Please, see the last commits

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/include/sdl_rpc_plugin/commands/hmi/on_device_connection_status_notification.h [hunk 18]
```diff
@@ -0,0 +1,129 @@
...
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef SRC_COMPONENTS_APPLICATION_MANAGER_INCLUDE_APPLICATION_MANAGER_COMMANDS_HMI_ON_DEVICE_CONNECTION_STATUS_NOTIFICATION_H_
```
[the remark this answers is not in the document]
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/include/sdl_rpc_plugin/commands/hmi/on_device_connection_status_notification.h]: fixed

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/get_device_connection_status_request.cc [hunk 19]
```diff
@@ -0,0 +1,218 @@
...
+#include "utils/helpers.h"
+
+namespace sdl_rpc_plugin {
+using namespace application_manager;
+namespace commands {
+
+/**
+ * @brief Functor used to check the
+ * availability of invalid characters
+ * in string as defined in APPLINK-28280
```
[the remark this answers is not in the document]
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/get_device_connection_status_request.cc]: fixed

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/get_device_connection_status_request.cc [hunk 20]
```diff
@@ -0,0 +1,218 @@
...
+                 "Failed to extract device handle for device: " << device_id);
+    return false;
+  }
+
+  const protocol_handler::SessionObserver& session_observer =
+      connection_handler.get_session_observer();
+  const int32_t invalid_result = -1;
+  if (invalid_result ==
+      session_observer.GetDataOnDeviceID(
+          device_handle, &out_device_name, NULL, NULL, NULL)) {
```
[the remark this answers is not in the document]
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/get_device_connection_status_request.cc]: fixed

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/get_system_info_request.cc [hunk 21]
```diff
@@ -62,3 +62,6 @@ void GetSystemInfoRequest::Run() {
 }  // namespace commands
 
 }  // namespace application_manager
+
+
+
```
[the remark this answers is not in the document]
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/get_system_info_request.cc]: fixed

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/hmi/get_device_connection_status_request_test.cc [hunk 22]
```diff
@@ -0,0 +1,443 @@
+/*
+ * Copyright (c) 2016, Ford Motor Company
```
[the remark this answers is not in the document]
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/hmi/get_device_connection_status_request_test.cc]: fixed

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/include/sdl_rpc_plugin/commands/hmi/get_device_connection_status_response.h [hunk 23]
```diff
@@ -0,0 +1,78 @@
...
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef SRC_COMPONENTS_APPLICATION_MANAGER_INCLUDE_APPLICATION_MANAGER_COMMANDS_HMI_GET_DEVICE_CONNECTION_STATUS_RESPONSE_H_
```
[the remark this answers is not in the document]
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/include/sdl_rpc_plugin/commands/hmi/get_device_connection_status_response.h]: fixed

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/include/sdl_rpc_plugin/commands/hmi/get_device_connection_status_request.h [hunk 24]
```diff
@@ -0,0 +1,160 @@
...
+   * @param key Key used to read the corresponding string value from device_data
+   * @param out_value Corresponding to key string value from device_data
+   * @param True if the out_value is valid
+   */
+  bool GetKeyStringValue(const smart_objects::SmartObject& device_data,
+                         const std::string& key,
+                         std::string& out_value) const;
+
+  /**
+   * @brief Check whether the string contains invalid characters
```
[the remark this answers is not in the document]
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/include/sdl_rpc_plugin/commands/hmi/get_device_connection_status_request.h]: fixed

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/include/sdl_rpc_plugin/commands/hmi/get_device_connection_status_request.h (hunk 16 above)
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/include/sdl_rpc_plugin/commands/hmi/get_device_connection_status_request.h]: fixed

File: src/components/application_manager/include/application_manager/policies/policy_handler.h (hunk 15 above)
Comment (ZhdanovP) [reply src/components/application_manager/include/application_manager/policies/policy_handler.h]: fixed

File: src/components/application_manager/include/application_manager/policies/policy_handler.h (hunk 14 above)
Comment (ZhdanovP) [reply src/components/application_manager/include/application_manager/policies/policy_handler.h]: fixed

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/include/sdl_rpc_plugin/commands/hmi/get_device_connection_status_request.h (hunk 13 above)
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/include/sdl_rpc_plugin/commands/hmi/get_device_connection_status_request.h]: fixed

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/include/sdl_rpc_plugin/commands/hmi/get_device_connection_status_request.h [hunk 25]
```diff
@@ -0,0 +1,160 @@
...
+
+  /**
+   * @brief Retrieves the corresponding to data device_id
+   * from Policy Table and stores in the index of the
+   * "device" section of the message
+   * @param device_id ID of the device in the Policy Table
+   * @param index Index position in "device" section of the message
+   * to store data.
+   */
+  void FillDataForDeviceAtIndex(const std::string& device_id, size_t index);
```
[the remark this answers is not in the document]
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/include/sdl_rpc_plugin/commands/hmi/get_device_connection_status_request.h]: fixed

File: src/components/application_manager/src/message_helper/message_helper.cc (hunk 9 above)
Comment (ZhdanovP) [reply src/components/application_manager/src/message_helper/message_helper.cc]: fixed

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/get_device_connection_status_request.cc (hunk 3 above)
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/get_device_connection_status_request.cc]: fixed

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/get_device_connection_status_request.cc (hunk 2 above)
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/get_device_connection_status_request.cc]: fixed

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/include/sdl_rpc_plugin/commands/hmi/get_device_connection_status_request.h [hunk 26]
```diff
@@ -0,0 +1,163 @@
...
+   **/
+  void Run() OVERRIDE;
+
+ private:
+  DISALLOW_COPY_AND_ASSIGN(GetDeviceConnectionStatusRequest);
+
+  /**
+   * @brief Reads the mandatory parameters
+   * as defined in APPLINK-28267
+   * On successfull
```
[the remark this answers is not in the document]
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/include/sdl_rpc_plugin/commands/hmi/get_device_connection_status_request.h]: fixed

**Comment by ZhdanovP** [comment]
@AByzhynar @AKalinich-Luxoft please take a look

File: src/components/application_manager/include/application_manager/commands/command_impl.h [hunk 27]
```diff
@@ -138,6 +138,14 @@ class CommandImpl : public Command {
...
+   * @param str - string that need to be checked
+   * @param allow_empty_string if true methods allow empty sting
+   * @return true if success otherwise return false
+   */
+  bool CheckSyntax(const std::string& str, bool allow_empty_line = false) const;
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/commands/command_impl.h]: @ZhdanovP propose to move this function to MessageHelper, make it static. 
And make optional argument as enum, but not Boolean. Also all couple of unit tests explicitly for this function

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/hmi_command_factory.cc (hunk 7 above)
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/hmi_command_factory.cc]: @AKalinich-Luxoft please take a look the latest commits.

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/hmi/get_device_connection_status_request_test.cc (hunk 17 above)
[the remark this answers is not in the document]
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/hmi/get_device_connection_status_request_test.cc]: @AByzhynar please take a look 96aab482d7452e9d7f79df8aea458f270cd203a9

File: src/components/application_manager/include/application_manager/commands/command_impl.h (hunk 27 above)
Comment (ZhdanovP) [reply src/components/application_manager/include/application_manager/commands/command_impl.h]: @LuxoftAKutsan I think now the CheckSyntax is convenient for use. Almost all commands use CheckSyntax internal.
Moreover If we will move it to MessageHelper, a lot of files with changes will be. I think this changes will be better to provide in separate PR if you insist on this changes.

Comment (LuxoftAKutsan) [reply src/components/application_manager/include/application_manager/commands/command_impl.h]: @ZhdanovP  ok , but any way please check unit tests for it

File: src/components/application_manager/src/commands/command_impl.cc (hunk 8 above)
Comment (LuxoftAKutsan) [reply src/components/application_manager/src/commands/command_impl.cc]: @AByzhynar @ZhdanovP  CommandImpl::CheckSyntax should not change internals of command so it should be conts

**Comment by dboltovskyi** [comment]
Rebuild required

**Comment by iCollin** [comment]
Closed in favor of https://github.com/smartdevicelink/sdl_core/pull/3369
